client/question/services/FeedbackGeneratorService.js

```diff
@@ -53,6 +61,57 @@ tie.factory('FeedbackGeneratorService', [
     var consecutiveSameRuntimeErrorCounter = 0;
 
     /**
+     * Object containing variations on feedback text for different scenarios.
+     *
+     * $type {Object.<string, Array.<string>>}
+     */
+    var correctnessFeedbackText = {
+      'input_to_try': [
+        'Would your code work for the following input?',
+        'How about the following input? Would your code still work?',
+        'Would your code work with the following?',
+        'Consider the input below. How would you code handle it?',
+        'Have you considered input such as the following?'
+      ],
+      'expected_output': [
+        ('Below is the output your code should produce for the given input. ' +
+         'Can you find the bug?'),
+        'Consider the input/output pair below. Can you find the bug?',
+        'Here is the input/output pair. Where could the bug be?',
+        ('Your code should produce the output shown below. Can you update ' +
+         'your code to produce the same output?'),
+        ('It looks like there is still a bug. Can you modify your code so ' +
+         'that it produces the output shown below?')
+      ],
+      'output_enabled': [
+        ('If you are really stuck, you can display the output of your code.'),
+        ('If you are stuck and need help, you can display the output of your ' +
+         'code.'),
+        ('If you can\'t seem to get unstuck, you can display the output of ' +
+         'your code.'),
+        ('If you feel stumped, you can display the output of your code.'),
+        ('If you can\'t find the bug, you can display the output of your code.')
+      ]
+    };
+
+    /**
+     * Object to hold index of correctness feedback given (values), per
+     * test suite (keys). Used when avoiding presenting the same feedback text
+     * consecutively.
+     *
+     * $type {Object.<string, number>}
+     */
+    var lastCorrectnessFeedback = {};
+
+    /**
+     * Index of last test suite for which corrective feedback was given. Used
```

Comment:
Systematize terminology -- call it either "correctness feedback" or "corrective feedback", but not both in different places. I think I lean towards the former, since that's what we've historically been using.

Reply:
Yep, another typo (I think I was sleepy when writing some of this :-P). I meant "correctness". Will do.